### PR TITLE
feat(dev)!: move dev build and local prod build workflow to pages js

### PR DESCRIPTION
**Pull request by benlife5**
Created simplified start commands:
`pages dev` : start a local development server
`pages prod`: start a local production server

**Modifications**
The modified `pages dev` adds `pages generate features` (generates a `features.json` file) and `yext pages generate-test-data` (generating test data in the `localData` directory) to the current flow. The previous dev build command in the starter repo ran `yext pages build` but only `yext pages generate-test-data` is necessary.

`pages prod` runs `yext pages build`, `yext pages render`, and `yext pages serve`. The previous prod build command in the starter repo ran `yext pages generate-test-data` but that is included in the `yext pages build` command.

To support these actions, a util called `runSubprocess` has been created to run shell commands asynchronously and stream their output to the main process's stdout. Basic test cases are also included.

**Command Line Options**
Flags have been added to enable/disable steps of the build process:
For `pages dev`, there is `--noGenFeatures` and `--noGenTestData`
For `pages prod`, there is `--noBuild` and `--noRender`

**Testing**
`pnpm run test` passes all tests
The starter repo runs as expected for both `npm run prod` and `npm run dev` with the updated module installed.

### CODE REVIEW

File: packages/pages/src/dev/dev.ts [hunk 1]
```diff
@@ -3,31 +3,102 @@ import { CommandModule } from "yargs";
 import open from "open";
 import { ProjectFilepaths } from "../common/src/project/structure.js";
 import { devServerPort } from "./server/middleware/constants.js";
+import runSubProcess from "../util/runSubprocess.js";
 
 interface DevArgs extends Pick<ProjectFilepaths, "scope"> {
   local?: boolean;
   "prod-url"?: boolean;
   "open-browser": boolean;
+  scope?: string;
```
Comment (mkilpatrick) [inline packages/pages/src/dev/dev.ts:12]: You don't need most of these. We don't want to use pages dev as a passthrough of all flags that need to be passed to the underlying calls. We will have our own defaults and if a user needs to do something else then they can call that command themselves.

File: packages/pages/src/prod/prod.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import runSubprocess from "../util/runSubprocess.js";
+
+interface DevArgs extends Pick<ProjectFilepaths, "scope"> {
+  root?: string;
+  port?: number;
+  files?: string;
+  output?: string;
+  localData?: string;
+  hostname?: string;
+  noBuild?: boolean;
```
Comment (mkilpatrick) [inline packages/pages/src/prod/prod.ts:12]: Same comment. I think the only think we need is noBuild and noRender.

**Review by mkilpatrick — COMMENTED** [review]
Nice work! That was more straightforward than I thought.

File: packages/pages/src/util/runSubprocess.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import process from "process";
+
+/**
+ * Creates a child process and writes to stdout and stderr in real time
+ * @param {String} command the CLI command to run
+ * @param {ReadonlyArray<string>} args the CLI arguments
+ * @return {Promise<number>} the process's exit code
+ */
+function runSubProcess(command: string, args: ReadonlyArray<string>) {
+  process.stdout.write("> " + command + " " + args.join(" ") + "\n");
```
[the remark this answers is not in the document]
Comment (benlife5) [reply packages/pages/src/util/runSubprocess.ts:11]: It literally prints as `> ` ... the yext CLI seems to use it to show something is being run?

```
This is a beta version of the Yext Command Line Interface

Please upgrade the Yext CLI. Current version: 0.1_340. Top version: 0.1_342.

In order to upgrade, please run `brew upgrade yext`


> pages-starter-react-locations@0.0.0 build:local
> pages build

vite v3.2.7 building for production...
```

I was using it to indicate in the logs that another command was running, but can remove or replace with something eles

File: packages/pages/src/util/runSubprocess.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { spawn } from "child_process";
+import process from "process";
+
+/**
+ * Creates a child process and writes to stdout and stderr in real time
+ * @param {String} command the CLI command to run
+ * @param {ReadonlyArray<string>} args the CLI arguments
+ * @return {Promise<number>} the process's exit code
+ */
+function runSubProcess(command: string, args: ReadonlyArray<string>) {
```
[the remark this answers is not in the document]
Comment (benlife5) [reply packages/pages/src/util/runSubprocess.ts:10]: How do I test on Windows?

[the remark this answers is not in the document]
Comment (benlife5) [reply packages/pages/src/util/runSubprocess.ts:10]: pages dev and pages prod confirmed to work on Windows

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**